pydis_site/apps/api/tests/test_reminders.py

```diff
@@ -0,0 +1,139 @@
+from datetime import datetime
+
+from django_hosts.resolvers import reverse
+
+from .base import APISubdomainTestCase
+from ..models import Reminder, User
+
+
+class UnauthedReminderAPITests(APISubdomainTestCase):
+    def setUp(self):
+        super().setUp()
+        self.client.force_authenticate(user=None)
+
+    def test_list_returns_401(self):
+        url = reverse('bot:reminder-list', host='api')
+        response = self.client.get(url)
+
+        self.assertEqual(response.status_code, 401)
+
+    def test_create_returns_401(self):
+        url = reverse('bot:reminder-list', host='api')
+        response = self.client.post(url, data={'not': 'important'})
+
+        self.assertEqual(response.status_code, 401)
+
+    def test_delete_returns_401(self):
+        url = reverse('bot:reminder-detail', args=('1234',), host='api')
+        response = self.client.delete(url)
+
+        self.assertEqual(response.status_code, 401)
+
+
+class EmptyDatabaseReminderAPITests(APISubdomainTestCase):
+    def test_list_all_returns_empty_list(self):
+        url = reverse('bot:reminder-list', host='api')
+        response = self.client.get(url)
+
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.json(), [])
+
+    def test_delete_returns_404(self):
+        url = reverse('bot:reminder-detail', args=('1234',), host='api')
+        response = self.client.delete(url)
+
+        self.assertEqual(response.status_code, 404)
+
+
+class ReminderCreationTests(APISubdomainTestCase):
+    def setUp(self):
+        super().setUp()
+        self.author = User.objects.create(
+            id=1234,
+            name='Mermaid Man',
+            discriminator=1234,
+            avatar_hash=None,
+        )
+        self.data = {
+            'author': self.author.id,
+            'content': 'Remember to...wait what was it again?',
+            'expiration': datetime.utcnow().isoformat(),
+            'jump_url': "https://www.google.com",
+            'channel_id': 123,
+        }
+        url = reverse('bot:reminder-list', host='api')
+        response = self.client.post(url, data=self.data)
+        self.assertEqual(response.status_code, 201)
+
+    def test_reminder_in_full_list(self):
+        url = reverse('bot:reminder-list', host='api')
+        response = self.client.get(url)
+
+        self.assertEqual(response.status_code, 200)
+
+        self.data['expiration'] += 'Z'  # Massaging a quirk of the response time format
+        self.data['active'] = True
+        self.data['id'] = 1
+        self.assertEqual(response.json(), [self.data])
```

Comment:
This isn't really a creation test. If you want to ensure the object exists, I suggest doing an assertion for it in the actual creation test:

```py
self.assertIsNotNone(Reminder.objects.filter(id=...).first())
```

Reply:
This is another case of using examples from the existing code base. I like your suggestion better - I didn't consider using the models directly rather than trying to shoe-horn everything into the API endpoints.